lib/membrane_webm/parser.ex

```diff
@@ -0,0 +1,291 @@
+    <<keyframe::1, reserved::3, invisible::1, lacing::2, discardable::1>> = flags
+
+    if reserved != 0 do
+      IO.puts("SimpleBlock reserved bits in header flag should all be 0 but they are #{reserved}")
+    end
+
+    lacing =
+      case lacing do
+        0b00 -> :no_lacing
+        0b01 -> :Xiph_lacing
+        0b11 -> :EBML_lacing
+        0b10 -> :fixed_size_lacing
+      end
+
+    # TODO deal with lacing != 00 https://tools.ietf.org/id/draft-lhomme-cellar-matroska-04.html#laced-data-1
+
+    %{
+      track_number: track_number_vint.vint_data,
+      timecode: timecode,
+      header_flags: %{
+        keyframe: keyframe == 1,
+        reserved: reserved,
+        invisible: invisible == 1,
+        lacing: lacing,
+        discardable: discardable == 1
+      },
+      data: data
+    }
+  end
+
+  def parse(bytes, :binary, _name) do
+    bytes
+    # Base.encode16(bytes)
+  end
+
+  def parse(bytes, :string, :CodecID) do
+    # Video	    “V_*”
+    # Audio	    “A_*”
+    # Subtitle  “S_*”
+    # Button	  “B_*”
+    codec_string = Enum.join(for <<c::utf8 <- bytes>>, do: <<c::utf8>>)
+
+    case codec_string do
+      "A_OPUS" -> :opus
+      "A_VORBIS" -> :vorbis
+      "V_VP8" -> :vp8
+      "V_VP9" -> :vp9
+      _ -> raise "Invalid codec: #{codec_string}"
+    end
+  end
+
+  def parse(bytes, :string, _name) do
+    chars = for <<c::utf8 <- bytes>>, do: <<c::utf8>>
+    chars |> Enum.take_while(fn c -> c != <<0>> end) |> Enum.join()
+  end
+
+  def parse(bytes, :utf_8, _name) do
+    bytes
+    |> String.codepoints()
+    |> Enum.reduce("", fn codepoint, result ->
+      <<parsed::8>> = codepoint
+      if parsed == 0, do: result, else: result <> <<parsed>>
+    end)
+  end
+
+  def parse(<<>>, :date, _name) do
+    {{2001, 1, 1}, {0, 0, 0}}
+  end
+
+  def parse(<<nanoseconds::big-signed>>, :date, _name) do
+    seconds_zero = :calendar.datetime_to_gregorian_seconds({{2001, 1, 1}, {0, 0, 0}})
+    seconds = div(nanoseconds, Time.nanosecond()) + seconds_zero
+    :calendar.gregorian_seconds_to_datetime(seconds)
+  end
+
+  def parse(bytes, :void, _name) do
+    # Base.encode16(bytes)
+    byte_size(bytes)
+  end
+
+  def parse(bytes, :unknown, _name) do
+    Base.encode16(bytes)
+  end
+
+  def parse(bytes, :ignore, _name) do
+    Base.encode16(bytes)
+  end
+
+  def parse_element(bytes) do
+    %{vint: vint, rest: bytes} = Vint.parse(bytes)
+    id = vint.element_id
+    %{vint: vint, rest: bytes} = Vint.parse(bytes)
+    data_size = vint.vint_data
+    # TODO: deal with unknown data size
+    {name, type} = Schema.classify_element(id)
+
+    if name == :Unknown do
+      IO.puts("unknown element ID: #{id}")
+    end
+
+    with %{bytes: data, rest: bytes} <- trim_bytes(bytes, data_size) do
+      element = {
+        name,
+        parse(data, type, name)
+      }
+
+      %{element: element, rest: bytes}
+    end
+  end
+
+  def trim_bytes(bytes, how_many) do
+    <<bytes::binary-size(how_many), rest::binary>> = bytes
```

Comment:
And if not then something like this will suffice:
```elixir
  def split_at(binary, n) do
    <<part::binary-size(n), rest::binary>> = binary

    {part, rest}
  end
```

And don't create maps if not necessary